fronts-client/src/services/recipeQuery.ts

```diff
@@ -0,0 +1,167 @@
+import url from '../constants/url';
+import { Recipe } from '../types/Recipe';
+
+interface KeyAndCount {
+  key: string;
+  doc_count: number;
+}
+
+export interface ChefSearchParams {
+  query?: string;
+  limit?: number;
+}
+
+export interface RecipeSearchFilters {
+  diets?: string[];
+  contributors?: string[];
+  filterType: "During" | "Post";
+}
+
+export interface RecipeSearchParams {
+  queryText: string;
+  searchType?: "Embedded"|"Match"|"Lucene",
+  fields?: string[];
+  kfactor?: number;
+  limit?: number;
+  filters?: RecipeSearchFilters;
+}
+
+export interface ChefSearchResponse {
+
+}
+
+export interface RecipeSearchResponse {
+  hits: number;
+  maxScore: number;
+  recipes: RecipeSearchHit[];
+}
+
+interface RecipeSearchTitlesResponse {
+  score: number;
+  href: string;
+}
+
+export type RecipeSearchHit = Recipe & {
+  score: number;
+}
+
+export interface DietSearchResponse {
+  "diet-ids": KeyAndCount[];
+}
+
+const widthParam = /width=(\d+)/;
+export const updateImageScalingParams = (url:string) => {
+  return url.replace(widthParam, "width=83")
+}
+
+const setupRecipeThumbnails = (recep:Recipe) => {
+  try {
+    return {
+      ...recep,
+      previewImage: recep.previewImage ? {
+        ...recep.previewImage,
+        url: updateImageScalingParams(recep.previewImage.url),
+      } : undefined,
+      featuredImage: recep.featuredImage ? {
+        ...recep.featuredImage,
+        url: updateImageScalingParams(recep.featuredImage.url),
+      } : undefined
+    }
+  } catch(err) {
+    console.error(err);
+    return recep;
+  }
+}
+
+const recipeQuery = (baseUrl:string) => {
+  const fetchOne = async (href:string):Promise<Recipe|undefined> => {
+    const response = await fetch(`${baseUrl}${href}`);
+    const content = await response.json();
```

Comment:
error handling or checking `response` we might need in case of `response` is of type of any error and not json?